document/js-api/index.bs

```diff
@@ -1470,25 +1473,80 @@ WebAssembly Module Records have the following methods:
+            1. Let |externfunc| be the [=external value=] [=external value|func=] |funcaddr|.
+            1. [=list/Append=] |externfunc| to |imports|.
+        1. If |importtype| is of the form [=global=] |mut| |valtype|,
+            1. Let |store| be the [=surrounding agent=]'s [=associated store=].
+            1. If |v| [=implements=] {{Global}},
+                1. Let |globaladdr| be |v|.\[[Global]].
+                1. Let |targetmut| <var ignore>valuetype</var> be [=global_type=](|store|, |globaladdr|).
+                    1. If |mut| is [=const=] and |targetmut| is [=var=], throw a {{LinkError}} exception.
+            1. Otherwise,
+                1. If |valtype| is [=v128=], throw a {{LinkError}} exception.
+                1. If |mut| is [=var=], throw a {{LinkError}} exception.
+                1. Let |value| be [=?=] [=ToWebAssemblyValue=](|v|, |valtype|).
+                1. Let (|store|, |globaladdr|) be [=global_alloc=](|store|, |mut| |valtype|, |value|).
+                1. Set the [=surrounding agent=]'s [=associated store=] to |store|.
+            1. Let |externglobal| be [=external value|global=] |globaladdr|.
+            1. [=list/Append=] |externglobal| to |imports|.
+        1. If |importtype| is of the form [=mem=] <var ignore>memtype</var>,
+            1. If |v| does not [=implement=] {{Memory}}, throw a {{LinkError}} exception.
+            1. Let |externmem| be the [=external value=] [=external value|mem=] |v|.\[[Memory]].
+            1. [=list/Append=] |externmem| to |imports|.
+        1. If |importtype| is of the form [=table=] <var ignore>tabletype</var>,
+            1. If |v| does not [=implement=] {{Table}}, throw a {{LinkError}} exception.
+            1. Let |tableaddr| be |v|.\[[Table]].
+            1. Let |externtable| be the [=external value=] [=external value|table=] |tableaddr|.
+            1. [=list/Append=] |externtable| to |imports|.
 1. [=Instantiate the core of a WebAssembly module=] |module| with |imports|, and let |instance| be the result.
-1. For each |name| in the [=export name list=] of |record|,
-    1. Perform ! |record|.\[[Environment]].InitializeBinding(|name|, ! Get(|instance|.\[[Exports]], |name|)).
+1. Set |record|.\[[Instance]] to |instance|.
+1. [=list/iterate|For each=] (|name|, |externtype|) of [=module_exports=](|module|),
+    1. If |externtype| is of the form [=global=] |mut| |globaltype|,
+        1. Assert: |externval| is of the form [=external value|global=] |globaladdr|.
+        1. Let [=external value|global=] |globaladdr| be |externval|.
+        1. Let |global_value| be [=global_read=](|store|, |globaladdr|).
+        1. If |globaltype| is not [=v128=],
+            1. Note: The condition above leaves unsupported JS values as uninitialized in TDZ and therefore as a reference error on
+                access. When integrating with shared globals, they may be excluded here similarly to v128 above.
```

Comment:
Why do we do this instead of simply not exposing them, making it a linking error?

Reply:
This is a great point and thanks for spotting this. The reason is that we need these bindings to work when Wasm modules import other Wasm modules. That is, while these values are not expressible in JS, they should still be exposed as exports to other Wasm. TDZ is therefore a useful way to represent this property.